src/libstd/net/tcp.rs

```diff
@@ -729,6 +729,7 @@ impl TcpListener {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn accept(&self) -> io::Result<(TcpStream, SocketAddr)> {
+        #[allow(unused_variables)]
```

Comment:
Can we add a comment, something like this? 

"In some configurations, e.g., WASM, TcpStream is not supported and hence this is dead code; suppress the resulting warning, since in other platforms, the code is not dead."

Using `#[allow]` in odd situations like this seems fine to me, but I think it should come with a comment. =)

Bonus points if we can use `#[cfg_attr]` to make it conditional.